system/gsettings.py

```diff
@@ -0,0 +1,248 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: gsettings
+short_description: Module to manage dconf settings.
+description:
+    - This module provides a python equivalent of the gsettings command-line tool
+    - for use in ansible.
+version_added: "1.9"
+notes:
+    - become and become_user must be used to specify the target configuration set.
+requirements: [ "PyGObject"]
+options:
+    settings:
+        description:
+            - A list of schema, key, value triplets to update.
+        required: true
+        default: None
+        version_added: 1.9
+author: Adrian Wood (@ag-wood)
+'''
+
+EXAMPLES = '''
+# Desktop settings for Ubuntu Unity Desktop:
+- hosts: all
+  gather_facts: no
+  tasks:
+    - name: Change Desktop configuration
+    become: yes
+    become_user: janedoe
+    gsettings:
+      settings:
+      # Ensure lock when screensaver activates
+      - schema: org.gnome.desktop.screensaver
+        key: lock-enabled
+        value: true
+      # Clean up the 'Dash' search widget
+      - schema: com.canonical.Unity.Lenses
+        key: disabled-scopes
+        value: ['more_suggestions-amazon.scope', 'more_suggestions-u1ms.scope',
+                'more_suggestions-populartracks.scope', 'music-musicstore.scope',
+                'more_suggestions-ebay.scope', 'more_suggestions-ubuntushop.scope',
+                'more_suggestions-skimlinks.scope']
+      # Power button invokes shutdown.
+      - schema: org.gnome.settings-daemon.plugins.power
+        key: button-power
+        value: 'shutdown'
+      # Desktop colours and wallpaper.
+      - schema: org.gnome.desktop.background
+        key: primary-color
+        value: '#000000'
+      - schema: org.gnome.desktop.background
+        key: secondary-color
+        value: '#000000'
+      - schema: org.gnome.desktop.background
+        key: picture-uri
+        value: 'file:///usr/share/backgrounds/Beach_by_Renato_Giordanelli.jpg'
+'''
+
+# Standard python libraries
+import os
+import pwd
+import time
+import signal
+import subprocess
+
+try:
+    from gi.repository import Gio, GLib
+except ImportError:
+    HAS_GIO = False
+else:
+    HAS_GIO = True
+
+ENV_DBUS = 'DBUS_SESSION_BUS_ADDRESS'
+SESSION_MANAGERS = [
+    'gnome-session', 'mate-session', 'xfce4-session',
+    'cinnamon-session', 'icewm-session', 'openbox-session']
+
+
+class Gsettings(object):
+    def __init__(self):
+        def proc_name(pid):
```

Comment:
It's a technique I've seen used to hide the implementation of functions that are only used inside the class function.  I'm very happy to move them if it's not the done thing. Still learning lots - thanks!